### PR TITLE
Remove assigned but never used local variables. Enable Travis checking for unused local variables.

### PR DESCRIPTION
**Pull request by practicalswift**
Remove assigned but never used local variables. Enable Travis checking for unused local variables.

### CODE REVIEW

**Comment by promag** [comment]
utACK 3f281a8.

**Comment by jnewbery** [comment]
Seems reasonable. On master:

```
→ flake8 --select F841 test/functional/feature_block.py 
test/functional/feature_block.py:661:9: F841 local variable 's' is assigned to but never used
test/functional/feature_block.py:698:9: F841 local variable 'b51' is assigned to but never used
test/functional/feature_block.py:816:9: F841 local variable 'b58' is assigned to but never used
test/functional/feature_block.py:823:9: F841 local variable 'b59' is assigned to but never used
test/functional/feature_block.py:864:9: F841 local variable 'b62' is assigned to but never used
test/functional/feature_block.py:1091:9: F841 local variable 'b74' is assigned to but never used
test/functional/feature_block.py:1104:9: F841 local variable 'b75' is assigned to but never used
test/functional/feature_block.py:1115:9: F841 local variable 'b76' is assigned to but never used
```

However, I've completely refactored this test in #11773 (which builds on #11771). My preference would be for those to be reviewed and merged.

I think removing all 'unused variable' warnings and enforcing F841 in our travis linter is a good longer term goal, but I'm concept NACKish on PRs that just fix style in individual files.

**Comment by WalterBoles** [comment]
Sent from Yahoo Mail for iPhone


On Monday, January 29, 2018, 7:58 AM, John Newbery <notifications@github.com> wrote:


I'm still NACKish. Rebasing #11773 on this will be a lot of wasted work.

F841 is a style issue that may hide bugs, but none of the changes here are actual bug fixes. I think it's a good long-term goal, but not worth opening individual PRs.

**Comment by practicalswift** [comment]
@jnewbery If I rebase on top of `master` once #11773 is merge, then this should be an ACK, no?

Assuming enforcing F841 in our Travis linter is a good longer term goal.

**Comment by jnewbery** [comment]
I agree that adding F841 should be a long-term goal.

> If I rebase this PR on top of master once #11773 is merged, then this should be an ACK, no?

Yes, but with the same caveats as #12295 - that I don't know the least disruptive way to fix lint warnings and add rules to Travis.

**Comment by practicalswift** [comment]
Rebased on top of `master` now that #12295 has been merged! :-)

**Comment by MarcoFalke** [comment]
I will look at this after #11773 is merged.

**Comment by practicalswift** [comment]
#11773 is now merged. Please re-review :-)

**Comment by MarcoFalke** [comment]
utACK 22d1bd1914651f3c00739652471aabf5374ef092

File: contrib/devtools/lint-python.sh [hunk 1]
```diff
@@ -22,6 +22,7 @@
 # E275 missing whitespace after keyword
 # E304 blank lines found after function decorator
 # E306 expected 1 blank line before a nested definition
+# F401: module imported but unused
```
Comment (jnewbery) [inline contrib/devtools/lint-python.sh]: This has snuck in somehow. Please remove! (F401 is documented lower down in the right place)

**Review by jnewbery — COMMENTED** [review]
utACK 22d1bd1914651f3c00739652471aabf5374ef092 after removing the stray comment in lint-python.sh

File: contrib/devtools/lint-python.sh (hunk 1 above)
Comment (practicalswift) [reply contrib/devtools/lint-python.sh]: Oh, thanks. Now fixed.

**Comment by practicalswift** [comment]
Updated. Please re-review :-)

**Comment by MarcoFalke** [comment]
utACK ea04bf786263eb0c933648bce43627c0de4d84ef

**Pull request merged**